packages/venia-concept/src/RootComponents/Product/Product.js

```diff
@@ -1,12 +1,71 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { Query } from 'react-apollo';
+import gql from 'graphql-tag';
 import { bool, shape, number, arrayOf, string } from 'prop-types';
 
 import { addItemToCart } from 'src/actions/cart';
 import ProductFullDetail from 'src/components/ProductFullDetail';
 import getUrlKey from 'src/util/getUrlKey';
-import getProductDetail from '../../queries/getProductDetail.graphql';
+
+const query = gql`
+    query productDetail($urlKey: String, $onServer: Boolean!) {
+        productDetail: products(filter: { url_key: { eq: $urlKey } }) {
+            items {
+                sku
+                name
+                price {
+                    regularPrice {
+                        amount {
+                            currency
+                            value
+                        }
+                    }
+                }
+                description
+                media_gallery_entries {
+                    label
+                    position
+                    disabled
+                    file
+                }
+                ... on ConfigurableProduct {
+                    configurable_options {
+                        attribute_code
+                        attribute_id
+                        id
+                        label
+                        values {
+                            default_label
+                            label
+                            store_label
+                            use_default_value
+                            value_index
+                        }
+                    }
+                    variants {
+                        product {
+                            fashion_color
+                            fashion_size
+                            id
+                            media_gallery_entries {
+                                disabled
+                                file
+                                label
+                                position
+                            }
+                            sku
+                            stock_status
+                        }
+                    }
```

Comment:
While it's necessary to fetch variants, this is expensive. It would be a big optimization to only fetch fewer variants.

Reply:
Sure. 👍